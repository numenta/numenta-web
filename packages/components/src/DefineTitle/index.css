/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.horizontal {
  composes: fl from 'tachyons';
  composes: h-100 from 'tachyons';
  composes: ma0 from 'tachyons';
  padding-bottom: 0.2rem;
  padding-top: 0.4rem;
  width: 35%;
}

.vertical {
  composes: lh-title from 'tachyons';
  composes: mb1 from 'tachyons';
}


.defineTitle {
  composes: fw6 from 'tachyons';
}
