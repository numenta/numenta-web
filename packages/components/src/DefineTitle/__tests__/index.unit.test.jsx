import React from 'react'
import renderer from 'react-test-renderer'

import DefineTitle from '../../DefineTitle'


describe('DefineTitle React component', () => {

  it('Renders Default correctly', () => {
    const component = renderer.create(
      <DefineTitle>
        Title 1
      </DefineTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Horizontally correctly', () => {
    const component = renderer.create(
      <DefineTitle direction="horizontal">
        Title 1
      </DefineTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Renders Vertically correctly', () => {
    const component = renderer.create(
      <DefineTitle direction="vertical">
        Title 1
      </DefineTitle>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
