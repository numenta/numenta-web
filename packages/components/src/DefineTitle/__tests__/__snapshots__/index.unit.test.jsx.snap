exports[`DefineTitle React component Renders Default correctly 1`] = `
<dt
  className="defineTitle">
  Title 1
</dt>
`;

exports[`DefineTitle React component Renders Horizontally correctly 1`] = `
<dt
  className="defineTitle horizontal">
  Title 1
</dt>
`;

exports[`DefineTitle React component Renders Vertically correctly 1`] = `
<dt
  className="defineTitle vertical">
  Title 1
</dt>
`;
